### PR TITLE
Polling cleanup, fix PHP warnings

### PR DESCRIPTION
**Pull request by murrant**
Capture all warnings generated by test data and fix them.
Mostly by converting os polling files to yaml.
Go ahead and convert most of the os polling files to yaml while I wait.

a lot of isset and null coalescing ??

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/Console/Commands/FindWarnings.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<?php
```
Comment (Jellyfrog) [inline app/Console/Commands/FindWarnings.php:1]: should we keep this? it is pretty after all :D

**Review by Jellyfrog — APPROVED** [review]
null coalescing is not pretty, but what to do :)

File: app/Console/Commands/FindWarnings.php (hunk 1 above)
Comment (murrant) [reply app/Console/Commands/FindWarnings.php:1]: I was on the fence.  I'll keep it for now and delete later, maybe. :)

File: tests/snmpsim/samsungprinter_clx.snmprec [hunk 2]
```diff
@@ -1 +1 @@
-1.3.6.1.2.1.1.1.0|4|Samsung CLX
+1.3.6.1.2.1.1.1.0|4|Samsung CLX-3170 Series;V1.00.01.64 Sep-27-2010;Engine 1.77.81;NIC V4.01.20(CLX-3170) 02-05-2010;S/N AAAAAAA000AAAA000
```
Comment (Jellyfrog) [inline tests/snmpsim/samsungprinter_clx.snmprec:1]: where is all this new data coming from? same with all other snmprec changes

File: tests/data/datacom.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
                 {
                     "sysName": "",
                     "sysObjectID": ".1.3.6.1.4.1.3709",
-                    "sysDescr": null,
+                    "sysDescr": "",
```
Comment (Jellyfrog) [inline tests/data/datacom.json:26]: should we make empty strings null?

File: tests/snmpsim/samsungprinter_clx.snmprec (hunk 2 above)
Comment (murrant) [reply tests/snmpsim/samsungprinter_clx.snmprec:1]: These are from the stats site.  It logs sysObjectID and sysDescr

File: tests/data/datacom.json (hunk 3 above)
Comment (murrant) [reply tests/data/datacom.json:26]: Probably should be null.  LibreNMS seems inconsistent with sys* variables and null/""

**Comment by glance-** [comment]
This will break newer areca cards, by just looking at 1.3.6.1.4.1.18928.1.1.1.1.0 and not 1.3.6.1.4.1.18928.1.2.1.1.0.

**Comment by murrant** [comment]
@glance- that is unrelated to this change.  If you want to add additional detections, please send a pull request :) `1.3.6.1.4.1.18928.1.2.1.1.0` was never used for hardware name before.

**Comment by glance-** [comment]
Yes, it was used:
https://github.com/librenms/librenms/blob/666638eeaabd0abb2f3d46d9cef6134cf55b9671/includes/polling/os/areca.inc.php#L5

The whole areca module is missing some vital functionality, like monitoring disks, raids and volumes but that have never bin there, and whenever I get time for it I'll try to make something up.

**Comment by murrant** [comment]
Sending a fix, next time create an issue instead of commenting on a merged PR ;)

**Pull request merged**